gobblin-runtime/src/main/java/gobblin/runtime/util/FSJobCatalogHelper.java

```diff
@@ -0,0 +1,402 @@
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.URISyntaxException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.commons.configuration.ConfigurationConverter;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+
+import com.typesafe.config.Config;
+import com.google.common.collect.Lists;
+import com.google.common.base.Splitter;
+import com.google.common.base.Charsets;
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.ImmutableSet;
+
+import gobblin.util.ConfigUtils;
+import gobblin.runtime.api.JobSpec;
+import gobblin.util.ResourceBasedTemplate;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.util.filesystem.PathAlterationListener;
+import gobblin.util.filesystem.PathAlterationDetector;
+import gobblin.util.filesystem.PathAlterationObserver;
+import gobblin.runtime.job_catalog.FSJobCatalog.Action;
+
+
+/**
+ * Pretty much the same as SchedulerUtils.java at current stage.
+ * Todo: To completely migrate those parts of code into refactored job launcher & scheduler .
+ */
+
+public class FSJobCatalogHelper {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FSJobCatalogHelper.class);
+
+  // Key used in the metadata of JobSpec.
+  private static final String DESCRIPTION_KEY_IN_JOBSPEC = "description";
+  private static final String VERSION_KEY_IN_JOBSPEC = "version";
+
+  /**
+   * Load job configuration from job configuration files stored in general file system,
+   * located by Path
+   * @param properties Gobblin framework configuration properties
+   *                   A good examplfied option of this is JOB_CONFIG_FILE_GENERAL_PATH_KEY
+   * @return a list of job configurations in the form of {@link JobSpec}
+   */
+  public static List<JobSpec> loadGenericJobConfigs(Properties properties)
+      throws ConfigurationException, IOException {
+    List<JobSpec> jobConfigs = Lists.newArrayList();
+    loadGenericJobConfigsRecursive(jobConfigs, properties, getJobConfigurationFileExtensions(properties),
+        new Path(properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_GENERAL_PATH_KEY)));
+    return jobConfigs;
+  }
+
+  /**
+   * Load a given job configuration file from a general file system.
+   *
+   * @param frameworkProps Gobblin framework configuration properties
+   * @param jobConfigRelativePath job configuration file to be loaded
+   * @param jobConfigDirPath The root job configuration directory path
+   * @return a single job configuration in the form of {@link JobSpec}
+   */
+  public static JobSpec loadGenericJobConfig(Properties frameworkProps, Path jobConfigRelativePath,
```

Comment:
Yes. There's no extension checking. 

Is the `*.properties` file to be deprecated ?